### PR TITLE
drivers: rtc: Add Zephyr RTC driver for NXP MCUX

### PR DESCRIPTION
**Pull request by frasa**
This patchset adds:
* Zephyr RTC driver for NXP MCUX
* DTS bindings for RTC
* DTS configuration for the FRDM-KW41Z board
* MKW41Z SoC Kconfiguration
* CMakeLists for compilation

and changes:
* `include/rtc.h` and `drivers/rtc/rtc_qmsi.c`, to remove clock divider enum replaced by Kconfig / DTS values
* RTC test
* RTC driver Kconfiguration, taking care of QMSI / MCUX separation

Find more information in the commit message bodies.

This was authored by @agross-korg, who is also available for change requests.

### CODE REVIEW

File: dts/bindings/rtc/rtc.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      type: int
+      category: optional
+      description: Clock frequency information for RTC operation
+      generation: define
+    label:
+      type: string
+      category: required
+      description: Human readable string describing the device (used by Zephyr for API name)
+      generation: define
+    interrupt-names:
```
Comment (MaureenHelm) [inline dts/bindings/rtc/rtc.yaml]: Why are interrupt-names part of the base RTC structure, but not interrupts?

File: boards/arm/frdm_kw41z/dts.fixup [hunk 2]
```diff
@@ -1,3 +1,7 @@
 #define CONFIG_FXOS8700_NAME			NXP_KINETIS_I2C_40067000_NXP_FXOS8700_1F_LABEL
 #define CONFIG_FXOS8700_I2C_NAME		NXP_KINETIS_I2C_40067000_NXP_FXOS8700_1F_BUS_NAME
 #define CONFIG_FXOS8700_I2C_ADDRESS		NXP_KINETIS_I2C_40067000_NXP_FXOS8700_1F_BASE_ADDRESS
+#define CONFIG_RTC_MCUX_0_IRQ_PRI		NXP_KINETIS_RTC_4003D000_IRQ_0_PRIORITY
+#define CONFIG_RTC_MCUX_0_IRQ			NXP_KINETIS_RTC_4003D000_IRQ_0
+#define CONFIG_RTC_MCUX_0_NAME			NXP_KINETIS_RTC_4003D000_LABEL
+#define CONFIG_RTC_PRESCALER			NXP_KINETIS_RTC_4003D000_PRESCALER
```
Comment (MaureenHelm) [inline boards/arm/frdm_kw41z/dts.fixup]: These changes should go in the soc-level fixup instead

File: drivers/rtc/rtc_mcux.c [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	config->irq_config_func(dev);
+	return 0;
+}
+
+static struct mcux_rtc_data rtc_mcux_data_0;
+
+static void rtc_mcux_irq_config_0(struct device *dev);
+
+static struct mcux_rtc_config rtc_mcux_config_0 = {
+	.base = RTC,
```
Comment (MaureenHelm) [inline drivers/rtc/rtc_mcux.c]: Why isn't this coming from dts?

File: dts/bindings/rtc/nxp,kinetis-rtc.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+---
```
Comment (MaureenHelm) [inline dts/bindings/rtc/nxp,kinetis-rtc.yaml]: Need to add license header

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

**Comment by frasa** [comment]
Change requests addressed, squashed and force-pushed.

File: dts/bindings/rtc/nxp,kinetis-rtc.yaml (hunk 4 above)
Comment (frasa) [reply dts/bindings/rtc/nxp,kinetis-rtc.yaml]: Now fixed in https://github.com/zephyrproject-rtos/zephyr/pull/7182/commits/0fc81773e63f439238ab803e04eb55aa90dec4ab

File: dts/bindings/rtc/rtc.yaml (hunk 1 above)
Comment (frasa) [reply dts/bindings/rtc/rtc.yaml]: Now fixed in https://github.com/zephyrproject-rtos/zephyr/pull/7182/commits/0fc81773e63f439238ab803e04eb55aa90dec4ab

File: boards/arm/frdm_kw41z/dts.fixup (hunk 2 above)
Comment (frasa) [reply boards/arm/frdm_kw41z/dts.fixup]: Fixed in https://github.com/zephyrproject-rtos/zephyr/pull/7182/commits/ec1a2be6d82313da7d5f761e8fda07c7d71808f6

File: drivers/rtc/rtc_mcux.c (hunk 3 above)
Comment (frasa) [reply drivers/rtc/rtc_mcux.c]: Fixed in https://github.com/zephyrproject-rtos/zephyr/pull/7182/commits/f17ac50ae56c2c3b2438c7e93c1cc97c51b56ccc

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**